src/NServiceBus.Core/Receiving/ReceiveCompleted.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Event raised when the transport has completed processing a message.
+    /// </summary>
+    public class ReceiveCompleted
+
+    {
+        /// <summary>
+        /// Constructs the event.
+        /// </summary>
+        public ReceiveCompleted(string messageId, bool wasAcknowledged, Dictionary<string, string> headers, DateTimeOffset startedAt, DateTimeOffset completedAt)
+        {
+            NativeMessageId = messageId;
+            WasAcknowledged = wasAcknowledged;
+            Headers = headers;
+            StartedAt = startedAt;
+            CompletedAt = completedAt;
+        }
+
+        /// <summary>
+        /// The native message ID.
+        /// </summary>
+        public string NativeMessageId;
```

Comment:
Are NativeMessageId and WasAcknowledged public _fields_ for a reason?

Reply:
No, will make them props